File: base_v4/IBaseView.java
package com.dev.rahul.basefilesamples.base_v4;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v4.widget.SwipeRefreshLayout;
import android.view.View;

/**
 * Created by rahul on 4/1/18.
 */

public interface IBaseView {

    Context getBaseContext();

    void enableBackButton();
    void disableBackButton();

    void setTitle(String title);
    void setSubTitle(String subTitle);

    void onAttachSwipeRefreshLayout(SwipeRefreshLayout swipeRefreshLayout);
    void setSwipeListener(SwipeRefreshLayout.OnRefreshListener listener);

    void onShowLoading();
    void onHideLoading();

    void showProgressDialog();
    void hideProgressDialog();

    void onFailed(@NonNull String message);
    void onError(@NonNull String error);
    void onSuccess(@NonNull String message);

    void initializeSnackBar(View view);
    void showSnackBarMessage(String message);

    void onAttachFragment(@NonNull Fragment fragment, @NonNull String tag);
    void onAttachFragment(@NonNull Fragment fragment, @NonNull String tag, boolean addToBackStack);
    void onDetachFragment(@NonNull String tag);
    void removeAllBackStackFragments();

    void hideKeyboard(View view);

    Bundle getBundleData();
}

File: SampleApp/basefiles/src/main/java/com/rp/basefiles/IBaseView.kt
package com.rp.basefiles

import android.os.Bundle
import android.view.View
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout


interface IBaseView {

    val bundleData: Bundle
    val isConnectedToNetwork: Boolean

    fun enableBackButton()
    fun disableBackButton()

    fun setTitle(title: String)
    fun setSubTitle(subTitle: String)

    fun onAttachSwipeRefreshLayout(swipeRefreshLayout: SwipeRefreshLayout)

    fun onShowLoading()
    fun onHideLoading()

    fun initializeSnackBar(view: View)
    fun showSnackBarMessage(message: String)

    fun hideKeyboard(view: View)
}